include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -596,10 +576,12 @@ struct __parallel_copy_if_static_single_group_submitter<_Size, _ElemsPerItem, _W
             // predicate on each element of the input range. The second half stores the index of the output
             // range to copy elements of the input range.
             auto __lacc = __dpl_sycl::__local_accessor<_ValueType>(sycl::range<1>{__elems_per_wg * 2}, __hdl);
-            auto __res_acc = __res.template get_access<access_mode::write>(__hdl);
+            auto __res_acc = __result.__get_result_acc(__hdl);
 
             __hdl.parallel_for<_ScanKernelName...>(
                 sycl::nd_range<1>(_WGSize, _WGSize), [=](sycl::nd_item<1> __self_item) {
+                    auto __res_ptr =
+                        __result_and_scratch_storage<_Policy, _Size>::__get_usm_or_buffer_accessor_ptr(__res_acc);
```

Comment:
At line 570 you wrote
`__result_and_scratch_storage<std::decay_t<_Policy>, _Size> __result{__policy, 0};`

but at this line `std::decay_t` is absent.

Which variant is correct?

Reply:
I think the decay is all unnecessary.  Working to confirm this and discussing with the others who contributed here who may have a better understanding of why this was introduced.

Reply:
I see that it's fixed.

Reply:
I've made the changes to remove the decays